src/AppInstallerCommonCore/Public/winget/WindowsFeature.h

```diff
@@ -0,0 +1,159 @@
+            DismLogErrorsWarningsInfoDebug
+        } DismLogLevel;
+
+        typedef enum _DismPackageIdentifier
+        {
+            DismPackageNone = 0,
+            DismPackageName,
+            DismPackagePath
+        } DismPackageIdentifier;
+
+        typedef enum _DismPackageFeatureState
+        {
+            DismStateNotPresent = 0,
+            DismStateUninstallPending,
+            DismStateStaged,
+            DismStateResolved, // For internal use only
+            DismStateRemoved = DismStateResolved,
+            DismStateInstalled,
+            DismStateInstallPending,
+            DismStateSuperseded,
+            DismStatePartiallyInstalled
+        } DismPackageFeatureState;
+
+        typedef enum _DismRestartType
+        {
+            DismRestartNo = 0,
+            DismRestartPossible,
+            DismRestartRequired
+        } DismRestartType;
+
+#pragma pack(push, 1)
+
+        typedef struct _DismCustomProperty
+        {
+            PCWSTR Name;
+            PCWSTR Value;
+            PCWSTR Path;
+        } DismCustomProperty;
+
+        typedef struct _DismFeature
+        {
+            PCWSTR FeatureName;
+            DismPackageFeatureState State;
+        } DismFeature;
+
+        typedef struct _DismFeatureInfo
+        {
+            PCWSTR FeatureName;
+            DismPackageFeatureState FeatureState;
+            PCWSTR DisplayName;
+            PCWSTR Description;
+            DismRestartType RestartRequired;
+            DismCustomProperty* CustomProperty;
+            UINT CustomPropertyCount;
+        } DismFeatureInfo;
+
+#pragma pack(pop)
+
+#ifdef __cplusplus
+    }
+#endif
+
+#endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_DISM) */
+#pragma endregion
+
+#endif // _DISMAPI_H_
+
+
+    struct DismApiHelper
+    {
+        DismApiHelper();
+        ~DismApiHelper();
+
+        DismFeatureInfo* GetFeatureInfo(const std::string_view& name);
+        HRESULT EnableFeature(const std::string_view& name);
+        HRESULT DisableFeature(const std::string_view& name);
+
+    private:
+        using DismInitializePtr = HRESULT(WINAPI*)(int, PCWSTR, PCWSTR);
+        using DismOpenSessionPtr = HRESULT(WINAPI*)(PCWSTR, PCWSTR, PCWSTR, UINT*);
+        using DismShutdownPtr = HRESULT(WINAPI*)();
+        using DismGetFeatureInfoPtr = HRESULT(WINAPI*)(UINT, PCWSTR, PCWSTR, DismPackageIdentifier, DismFeatureInfo**);
+        using DismEnableFeaturePtr = HRESULT(WINAPI*)(UINT, PCWSTR, PCWSTR, DismPackageIdentifier, BOOL, PCWSTR*, UINT, BOOL, HANDLE, DISM_PROGRESS_CALLBACK, PVOID);
+        using DismDisableFeaturePtr = HRESULT(WINAPI*)(UINT, PCWSTR, PCWSTR, BOOL, HANDLE, DISM_PROGRESS_CALLBACK, PVOID);
+        using DismDeletePtr = HRESULT(WINAPI*)(VOID*);
+
+        typedef UINT DismSession;
+
+        wil::unique_hmodule m_module;
+        DismSession m_session = DISM_SESSION_DEFAULT;
+        DismFeatureInfo* m_featureInfo = nullptr;

```

Comment:
This shouldn't be holding state that is feature specific.
You shouldn't be using a raw pointer if you can help it, use a smart pointer type instead.  Both `std::unique_ptr` and `wil::unique_any` can handle this.

Reply:
I tried using a unique_ptr but I don't think it's possible to access the memory address of a unique_ptr which is what I need for the **FeatureInfo parameter in [DismFeatureInfo](https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/dism/dismgetfeatureinfo-function?view=windows-11). I believe this is okay though because DismDelete should handle the cleanup of this pointer.